### PR TITLE
Add file filter to importer

### PR DESCRIPTION
**Pull request by carlreid**
In my case, my scanner likes to create a temporary file in the folder that is being watched by the importer. So in the current form, the importer than imports a garbage file.

This adds a whitelist filter (*I saw `recursive-readdir` has a blacklist approach built it but I thought that would suck to use*) using `minimatch`, which is a dependency of `recursive-readdir`, to allow for patterns like `*.+(pdf)` which I am using myself.  
  
If this value is left empty int he wizard then it defaults to the wildcard `*`.

### CODE REVIEW

**Comment by jendib** [comment]
@Cornelicorn Could you check this out? Thanks :)

File: docs-importer/main.js [hunk 1]
```diff
@@ -344,6 +364,8 @@ const start = () => {
 // Import the files
 const importFiles = (remove, filesImported) => {
   recursive(prefs.importer.path, function (error, files) {
+
+    files = files.filter(minimatch.filter(prefs.importer.fileFilter ?? "*", {matchBase: true}));
```
Comment (Cornelicorn) [inline docs-importer/main.js:368]: Haven't found documentation on `??` but seems to work

Comment (carlreid) [reply docs-importer/main.js:368]: Can read up here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

File: docs-importer/main.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 const recursive = require('recursive-readdir');
+const minimatch = require("minimatch");
```
[the remark this answers is not in the document]
Comment (carlreid) [reply docs-importer/main.js:4]: Do you think I should add the package even if it's a dependency of `recursive-readdir` anyway? I guess then it's clearer if I do.

File: docs-importer/main.js [hunk 3]
```diff
@@ -142,13 +143,32 @@ const askPath = () => {
...
 
+// Ask for the file filter
+const askFileFilter = () => {
+  console.log('');
+
+  inquirer.prompt([
+    {
+      type: 'input',
+      name: 'fileFilter',
+      message: 'What pattern do you want to use to match files? (eg. *.+(pdf|txt|jpg) - defaults to *)',
```
Comment (Cornelicorn) [inline docs-importer/main.js:161]: I dont think `- defaults to *` is necessary as the default value / saved value is displayed in parenthesis after the question

File: docs-importer/main.js [hunk 4]
```diff
@@ -142,13 +143,32 @@ const askPath = () => {
...
+      default: prefs.importer.fileFilter || "*"
+    }
+  ]).then(answers => {
+    // Save fileFilter
+    prefs.importer.fileFilter = answers.fileFilter  ? answers.fileFilter : '*';
```
Comment (Cornelicorn) [inline docs-importer/main.js:166]:
```suggestion
    prefs.importer.fileFilter = answers.fileFilter;
```
value is set in L162

Comment (carlreid) [reply docs-importer/main.js]: Makes complete sense, amended the previous commit.

**Pull request merged**